time_manager/include/time_manager.inc

```diff
@@ -18,1202 +18,46 @@
-!       Returns the double precision quotient of two times
-!   </OUT>
-
-!> Returns the double precision quotient of two times
-function time_real_divide(time1, time2)
-
-double precision :: time_real_divide
-type(time_type), intent(in) :: time1, time2
-double precision :: d1, d2
-
-if(.not.module_is_initialized) call time_manager_init
-
-! Convert time intervals to floating point seconds; risky for general performance?
-d1 = time1%days * dble(seconds_per_day) + dble(time1%seconds) + dble(time1%ticks)/dble(ticks_per_second)
-d2 = time2%days * dble(seconds_per_day) + dble(time2%seconds) + dble(time2%ticks)/dble(ticks_per_second)
-
-time_real_divide = d1 / d2
-
-end function time_real_divide
-! </FUNCTION>
-
-!-------------------------------------------------------------------------
-! <SUBROUTINE NAME="time_assignment; assignment(=)">
-
-!   <OVERVIEW>
-!       Assigns all components of the time_type variable on
-!       RHS to same components of time_type variable on LHS.
-!   </OVERVIEW>
-!   <DESCRIPTION>
-!       Assigns all components of the time_type variable on
-!       RHS to same components of time_type variable on LHS.
-!   </DESCRIPTION>
-!   <TEMPLATE>
-!     time1 = time2
-!   </TEMPLATE>
-
-!   <OUT NAME="time1" UNITS="" TYPE="time_type" DIM="">
-!      A time type variable.
-!   </OUT>
-!   <IN NAME="time2" UNITS="" TYPE="time_type" DIM="">
-!      A time type variable.
-!   </IN>
-
-!> Assigns all components of the time_type variable on
-!! RHS to same components of time_type variable on LHS.
-subroutine time_assignment(time1, time2)
-type(time_type), intent(out) :: time1
-type(time_type), intent(in)  :: time2
-   time1%seconds = time2%seconds
-   time1%days    = time2%days
-   time1%ticks   = time2%ticks
-end subroutine time_assignment
-! </SUBROUTINE>
-
-!-------------------------------------------------------------------------
-! <FUNCTION NAME="time_type_to_real">
-!   <OVERVIEW>
-!       Converts time to seconds and returns it as a real number
-!   </OVERVIEW>
-!   <DESCRIPTION>
-!       Converts time to seconds and returns it as a real number
-!   </DESCRIPTION>
-!   <TEMPLATE>
-!     time_type_to_real(time)
-!   </TEMPLATE>
-!   <IN NAME="time" UNITS="" TYPE="time_type" DIM="">
-!      A time interval.
-!   </IN>
-
-function time_type_to_real(time)
-
-real(kind=r8_kind)           :: time_type_to_real
-type(time_type), intent(in) :: time
-
-if(.not.module_is_initialized) call time_manager_init
-
-time_type_to_real = real(dble(time%days) * 86400.d0 + dble(time%seconds) + &
-     dble(time%ticks)/dble(ticks_per_second), kind=r8_kind)
-
-end function time_type_to_real
 
 !> @brief Convert a real number of seconds into a time_type variable.
 !! @return A filled time type variable, and an error message if an
 !!         error occurs.
-function real_to_time_type(x,err_msg) result(t)
-  real,intent(in) :: x !< Number of seconds.
+function REAL_TO_TIME_TYPE(x,err_msg) result(t)
```

Comment:
A general question, I'm having a hard time understanding this function.  Where do the components of t get assigned?